src/assets/scripts/game/GameController.js

```diff
@@ -68,107 +71,56 @@ export default class GameController {
-            arrival: 0,
-            departure: 0,
-
-            windy_landing: 0,
-            windy_takeoff: 0,
-
-            failed_arrival: 0,
-            failed_departure: 0,
-
-            warning: 0,
-            hit: 0,
-
-            abort: {
-                landing: 0,
-                taxi: 0
-            },
-
-            violation: 0,
-            restrictions: 0
-        };
-
-        prop.game.option = new GameOptions();
+        // Initialize event count
+        _forEach(GAME_EVENTS, (v, k) => {
+            prop.game.events[k] = 0;
+        });
     }
 
     /**
      * @for GameController
-     * @method game_get_score
+     * @method game_get_weighted_score
      */
-    game_get_score() {
-        let score = 0;
-
-        score += prop.game.score.arrival * 10;
-        score += prop.game.score.departure * 10;
-
-        score -= prop.game.score.windy_landing * 0.5;
-        score -= prop.game.score.windy_takeoff * 0.5;
-
-        score -= prop.game.score.failed_arrival * 20;
-        score -= prop.game.score.failed_departure * 2;
-
-        score -= prop.game.score.warning * 5;
-        score -= prop.game.score.hit * 50;
-
-        score -= prop.game.score.abort.landing * 5;
-        score -= prop.game.score.abort.taxi * 2;
-
-        score -= prop.game.score.violation;
-        score -= prop.game.score.restrictions * 10;
-
-        return score;
+    game_get_weighted_score() {
+        const hoursPlayed = this.game_time() / TIME.ONE_HOUR_IN_SECONDS;
+        const scorePerHour = this.game.score / hoursPlayed;
+        return scorePerHour;
     }
 
     /**
+     * Add points for a score event to the current game's score
      * @for GameController
-     * @method game_get_weighted_score
+     * @method game_add_to_score
+     * @param scoreEvent {String} one of the events listed in SCORE_EVENTS_VALUES
      */
-    game_get_weighted_score() {
-        let score = this.game_get_score();
-
-        score = score / (this.game_time() / 60);
-        score *= 500;
-
-        return score;
+    game_add_to_score(scoreEvent) {
+        prop.game.score[scoreEvent] += 1;
+        prop.game.score.points += SCORE_EVENTS_VALUES[scoreEvent];
     }
 
     /**
      * @for GameController
-     * @method game_reset_score
+     * @method game_reset_score_and_events
      */
-    game_reset_score() {
-        prop.game.score.abort = { landing: 0, taxi: 0 };
-        prop.game.score.arrival = 0;
-        prop.game.score.departure = 0;
-        prop.game.score.failed_arrival = 0;
-        prop.game.score.failed_departure = 0;
-        prop.game.score.hit = 0;
-        prop.game.score.restrictions = 0;
-        prop.game.score.violation = 0;
-        prop.game.score.warning = 0;
-        prop.game.score.windy_landing = 0;
-        prop.game.score.windy_takeoff = 0;
+    game_reset_score_and_events() {
+        // Reset events
+        _forEach(this.game.events, (v, k) => {
+            this.game.events[k] = 0;
+        });
```

Comment:
same as the other `_forEach` be sure to spell `v` and `k` out and provide contextual meaning to them.

👍 👍  for using `_forEach` with this new stuff, by the way.